src/plot_api/plot_api.js

```diff
@@ -322,9 +322,9 @@ Plotly.plot = function(gd, data, layout, config) {
     var donePlotting = Lib.syncOrAsync([
         Plots.previousPromises,
         marginPushers,
-        layoutStyles,
         marginPushersAgain,
         positionAndAutorange,
+        layoutStyles,
```

Comment:
`layoutStyles` is what calls the code that creates the `plotclip` - if it is called earlier, the value of `axis._length` is incorrect and the `plotclip` is too small. 

This [image diff](https://circle-artifacts.com/gh/plotly/plotly.js/996/artifacts/0/home/ubuntu/plotly.js/build/test_images_diff/diff-benchmarks.png) shows the improper clip. It's sort of hard to see, but the right side of the longest bars is clipped.